packages/flutter/lib/src/material/color_scheme.dart

```diff
@@ -8,163 +8,382 @@ import 'package:flutter/widgets.dart';
 import 'colors.dart';
 import 'theme_data.dart';
 
-/// A set of twelve colors based on the
-/// [Material spec](https://material.io/design/color/the-color-system.html)
+/// A set of colors based on the
+/// [Material spec](https://m3.material.io/styles/color/the-color-system/color-roles)
 /// that can be used to configure the color properties of most components.
 ///
 /// The [Theme] has a color scheme, [ThemeData.colorScheme], which is constructed
 /// with [ColorScheme.fromSwatch].
 @immutable
 class ColorScheme with Diagnosticable {
   /// Create a ColorScheme instance.
+  ///
+  /// For the color parameters that are nullable, it is still recommended
+  /// that applications provide values for them. They are only nullable due
+  /// to backwards compatibility concerns.
+  ///
+  /// If a color is not provided, the closest fallback color from the given
+  /// colors will be used for it (e.g. [primaryContainer] will default
+  /// to [primary]). Material 3 makes use of these colors for many component
+  /// defaults, so for the best results the application should supply colors
+  /// for all the parameters.
   const ColorScheme({
+    required this.brightness,
     required this.primary,
-    required this.primaryVariant,
-    required this.secondary,
-    required this.secondaryVariant,
-    required this.surface,
-    required this.background,
-    required this.error,
     required this.onPrimary,
+    Color? primaryContainer,
+    Color? onPrimaryContainer,
+    required this.secondary,
     required this.onSecondary,
-    required this.onSurface,
-    required this.onBackground,
+    Color? secondaryContainer,
+    Color? onSecondaryContainer,
+    Color? tertiary,
+    Color? onTertiary,
+    Color? tertiaryContainer,
+    Color? onTertiaryContainer,
+    required this.error,
     required this.onError,
-    required this.brightness,
-  }) : assert(primary != null),
-       assert(primaryVariant != null),
-       assert(secondary != null),
-       assert(secondaryVariant != null),
-       assert(surface != null),
-       assert(background != null),
-       assert(error != null),
+    Color? errorContainer,
+    Color? onErrorContainer,
+    required this.background,
+    required this.onBackground,
+    required this.surface,
+    required this.onSurface,
+    Color? surfaceVariant,
+    Color? onSurfaceVariant,
+    Color? outline,
+    Color? shadow,
+    Color? inverseSurface,
+    Color? onInverseSurface,
+    Color? inversePrimary,
+    @Deprecated(
```

Comment:
This PR should also include "dart fixes" for these properties. Something like the changes to packages/flutter/lib/fix_data.yaml in https://github.com/flutter/flutter/pull/87281/files. In this case you want to remove the places where the properties are used as constructor parameters, and change property references to primary or secondary.